tests/test_templatetags.py

```diff
@@ -356,3 +356,253 @@ def test_nested_component_templates_all_shown_as_used(self):
+
+    def test_both_slots_overridden(self):
+        template = Template(
+            """
+            {% load component_tags %}
+            {% component_block 'test' branch='a' %}
+                {% slot 'a' %}Override A{% endslot %}
+                {% slot 'b' %}Override B{% endslot %}
+            {% endcomponent_block %}
+            {% component_block 'test' branch='b' %}
+                {% slot 'a' %}Override A{% endslot %}
+                {% slot 'b' %}Override B{% endslot %}
+            {% endcomponent_block %}
+        """
+        )
+        rendered = template.render(Context({}))
+        self.assertHTMLEqual(rendered, '<p id="a">Override A</p><p id="b">Override B</p>')
+
+
+class SlotSuperTests(SimpleTestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        component.registry.clear()
+        component.registry.register('test', SlottedComponent)
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+        component.registry.clear()
+
+    def test_basic_super_functionality(self):
+        template = Template(
+            """
+            {% load component_tags %}
+            {% component_block "test" %}
+                {% slot "header" %}Before: {{ slot.super }}{% endslot %}
+                {% slot "main" %}{{ slot.super }}{% endslot %}
+                {% slot "footer" %}{{ slot.super }}, after{% endslot %}
+            {% endcomponent_block %}
+        """
+        )
+        rendered = template.render(Context({}))
+
+        self.assertHTMLEqual(
+            rendered,
+            """
+            <custom-template>
+                <header>Before: Default header</header>
+                <main>Default main</main>
+                <footer>Default footer, after</footer>
+            </custom-template>
+        """,
+        )
+
+    def test_multiple_super_calls(self):
+        template = Template(
+            """
+            {% load component_tags %}
+            {% component_block "test" %}
+                {% slot "header" %}First: {{ slot.super }}; Second: {{slot.super }}{% endslot %}
+            {% endcomponent_block %}
+        """
+        )
+        rendered = template.render(Context({}))
+
+        self.assertHTMLEqual(
+            rendered,
+            """
+            <custom-template>
+                <header>First: Default header; Second: Default header</header>
+                <main>Default main</main>
+                <footer>Default footer</footer>
+            </custom-template>
+        """,
+        )
+
+    def test_super_is_scoped_to_slot(self):
+        template = Template(
+            """
+            {% load component_tags %}
+            {% component_block "test" %}
+                {% slot "header" %}Override header{% endslot %}{{ slot.super }}
+                {% slot "main" %}{{ slot.super }}{% endslot %}{{ slot.super }}
+                {% slot "footer" %}{{ slot.super }}{% endslot %}{{ slot.super }}
```

Comment:
I think the behaviour here should be to raise an error instead of erroring silently?

Reply:
Makes sense and will do.

Reply:
Actually, unresolving this after spending more time with it.  This pattern is just a specific case of having any content in a component block that isn't also in a slot block.  It's actually hard to raise an error for bad `slot.super` calls specifically because anything that isn't a `SlotNode` is just discarded during parsing.  We'd have to inspect the skipped nodes and decide whether or not to raise the error.  Would it be better to treat any orphan nodes as `TemplateSyntaxError`s (either all the time or only in debug mode)?  Or debug mode warnings?

Reply:
I'm totally fine with leaving it as it is. I think it would be nicer, but not worth a lot of effort imho!

Reply:
Marking this as resolved!

Reply:
I ended up putting this in and playing around with the parsing code generally, but it's a separate commit that we can easily leave out if you prefer.